lib/middleware.ts

```diff
@@ -113,6 +113,97 @@ var defaultHandlers = [
   () => ({title: "request-finished"}),
 ];
 
+const defaultContextHandlers = [
+  (req) => ({"request-id": req.headers["x-request-id"]}),
+  (req) => ({"session-id": req.sessionID}),
+];
+
+function handlerData(handlers, ...args) {
+  const data = {};
+  handlers.forEach((h) => {
+    try {
+      const handler_data = h(...args);
+      _.extend(data, handler_data);
+    } catch (e) {
+      // ignore invalid handler
+    }
+  });
+  return data;
+}
+
+class ContextLogger {
```

Comment:
We could do something like:

```es6
['debug', 'info', 'warn', 'error', 'critical', 'counter', 'gauge'].forEach(func) {
  ContextLogger.prototype[func] = function (title) {
    this[func + 'D'](title, {});
  };
};
```
same for the `infoD` etc. Seems like are exposed  in the kayvee base package as Kayvee.LEVELS, maybe Kayvee.LEVELS + Kayvee.METRICS?